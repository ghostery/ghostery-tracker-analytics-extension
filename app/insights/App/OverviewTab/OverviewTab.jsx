import React from 'react';
import PropTypes from 'prop-types';

import TrackerDistributionGraph from '../../../shared-components/TrackerDistributionGraph';
import PageLatencyGraph from '../../../shared-components/PageLatencyGraph';
import PageSizeGraph from '../../../shared-components/PageSizeGraph';
import TrackerList from '../../../shared-components/TrackerList';
import './OverviewTab.scss';

const OverviewTab = ({
  trackerListExpanded, trackerDistributionExpanded, pageLatencyExpanded,
}) => {
  const topGraphExpanded = trackerDistributionExpanded || pageLatencyExpanded;
  return (
    <div className="d-flex flex-column align-items-center">
      {!trackerListExpanded && (
        <div className="OverviewTab__topGraphs d-flex align-items-center justify-content-between">
          {(!topGraphExpanded || trackerDistributionExpanded) && (
            <TrackerDistributionGraph
              expanded={trackerDistributionExpanded}
              width={trackerDistributionExpanded ? 950 : 300}
              height={300}
            />
          )}
          {(!topGraphExpanded || pageLatencyExpanded) && (
            <PageLatencyGraph
              expanded={pageLatencyExpanded}
              width={pageLatencyExpanded ? 950 : 350}
              height={300}
            />
          )}
          {!topGraphExpanded && (
            <PageSizeGraph height={300} width={250} />
          )}
        </div>
      )}
      <div className="OverviewTab__trackerTimeline">
        <TrackerList trackerListExpanded={trackerListExpanded} showHeader showGraph />
      </div>
    </div>
  );
};

OverviewTab.propTypes = {
  trackerListExpanded: PropTypes.bool.isRequired,
  trackerDistributionExpanded: PropTypes.bool.isRequired,
  pageLatencyExpanded: PropTypes.bool.isRequired,
};

export default OverviewTab;
