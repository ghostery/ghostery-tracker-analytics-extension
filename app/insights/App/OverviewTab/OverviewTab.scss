.OverviewTab__topGraphs {
  padding: 25px 40px;
  width: 954px;
}

.OverviewTab__trackerTimeline {
  padding-bottom: 25px;
  width: 950px;
}
