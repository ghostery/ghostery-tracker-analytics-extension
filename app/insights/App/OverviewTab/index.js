import { connect } from 'react-redux';

import OverviewTab from './OverviewTab.jsx';

const mapStateToProps = state => ({
  trackerListExpanded: state.UserInterfaceReducer.expandedComponents.trackerList,
  trackerDistributionExpanded: state.UserInterfaceReducer.expandedComponents.trackerDistribution,
  pageLatencyExpanded: state.UserInterfaceReducer.expandedComponents.pageLatency,
});

export default connect(mapStateToProps)(OverviewTab);
