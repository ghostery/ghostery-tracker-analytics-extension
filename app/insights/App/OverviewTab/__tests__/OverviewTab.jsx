import React from 'react';
import { shallow } from 'enzyme';
import path from 'path';
import fakeProps from 'react-fake-props';

import OverviewTab from '../OverviewTab';

jest.mock('../../../../../src/classes/Globals', () => ({
  BROWSER_INFO: { displayName: 'Chrome' },
}));

describe('app/insights/App/OverviewTab', () => {
  describe('Snapshot tests with react-test-renderer', () => {
    test('OverviewTab component should render', () => {
      const componentPath = path.join(__dirname, '../OverviewTab.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<OverviewTab {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
