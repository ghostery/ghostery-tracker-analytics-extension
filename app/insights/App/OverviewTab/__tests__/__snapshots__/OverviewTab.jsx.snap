// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/insights/App/OverviewTab Snapshot tests with react-test-renderer OverviewTab component should render 1`] = `
<div
  className="d-flex flex-column align-items-center"
>
  <div
    className="OverviewTab__trackerTimeline"
  >
    <Connect(TrackerList)
      showGraph={true}
      showHeader={true}
      trackerListExpanded={true}
    />
  </div>
</div>
`;
